File: src/main/java/tripway/server/service/users/UserService.java
package tripway.server.service.users;

import tripway.server.models.User;

import java.util.List;

/**
 * @author Nikita Burtelov
 */
public interface UserService {
    User getUserId(Long id);

    List<User> getAllUser();

    void saveUser(User user);

    void removeUser(User user);

    void removeUser(Long id);

    void updateUser(User user);

    void updateUser(Long id);
}

File: src/main/java/tripway/server/service/users/UserService.kt
package tripway.server.service.users

import tripway.server.models.UserEntity

interface UserService {
    fun getUserById(id: String): UserEntity
    val allUser: List<UserEntity>
    fun saveUser(user: UserEntity)
    fun removeUser(user: UserEntity)
    fun removeUser(id: String)
    fun updateUser(user: UserEntity)
    fun updateUser(id: String)
    fun subscribeUser(fromUserId: String, toUserId: String): Int
    fun unsubscribeUser(fromUserId: String, toUserId: String): Int
    fun getSubscribersCount(uid: String): Int
    fun getSubscriptionsCount(uid: String): Int
    fun isSubscribed(from: String, to: String): Boolean
}
